src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.CompilationTrackerState.cs

```diff
@@ -76,11 +76,11 @@ private sealed class InProgressState : CompilationTrackerState
                 /// The list of changes that have happened since we last computed a compilation. The oldState corresponds to
                 /// the state of the project prior to the mutation.
                 /// </summary>
-                public ImmutableList<(ProjectState oldState, CompilationAndGeneratorDriverTranslationAction action)> PendingTranslationSteps { get; }
+                public ImmutableList<TranslationAction> PendingTranslationActions { get; }
```

Comment:
1. i always found it unpleasant to hve to carry around this tuple and keep in sync.  So i pushed `ProjectState oldState` into `CompilationAndGeneratorDriverTranslationAction`
2. Rename CompilationAndGeneratorDriverTranslationAction to `TranslationAction`.